unittests/test_modules.py

```diff
@@ -199,6 +200,7 @@ class TestModuleMapping(unittest.TestCase):
     def setUp(self):
         self.modules_activity = ModulesSystemEmulator()
         self.modules_system = modules.ModulesSystem(self.modules_activity)
+        self.fp = NamedTemporaryFile('wt', delete=False)
```

Comment:
Here you should not use `fp`. It's like using `i` as a variable outside the scope of a for loop in C/C++... Better call it sth more meaningful, e.g., `mapping_file`.